engine/graphics/image.cpp

```diff
@@ -0,0 +1,56 @@
+#include "image.h"
+
+#include "utils.h"
+
+namespace pong
+{
+
+namespace image
+{
+
+/* Converts an alpha image to RGBA format. */
+std::vector<unsigned char> ConvertAlphaImageToRGBA(const std::vector<unsigned char>& imageData)
+{
+    std::vector<unsigned char> convertedImage {};
+    convertedImage.reserve(imageData.size() * 4);
+
+    const unsigned char kFullColor = 255;
+    for (auto& alpha : imageData)
+    {
+        convertedImage.emplace_back(kFullColor);
+        convertedImage.emplace_back(kFullColor);
+        convertedImage.emplace_back(kFullColor);
+        convertedImage.emplace_back(alpha);
```

Comment:
Not sure if it makes any difference here, but for simple objects (like a byte of unsigned char) `push_back` and `emplace_back` do the same thing. Not sure if there is a performance cost to `emplace`

Reply:
Not a confident answer I have but it seems like the compiler will probably be smart enough to optimize it away, but `push_back` might even be a bit safer and more readable.
https://stackoverflow.com/questions/48158286/emplace-back-vs-push-back-for-primitive-types

Reply:
Cool! Today I learned.

Seems like either is fine